pkg/query-service/app/http_handler.go

```diff
@@ -2669,6 +2668,20 @@ func (aH *APIHandler) getLogFieldsV3(ctx context.Context, queryRangeParams *v3.Q
 	return data, nil
 }
 
+func (aH *APIHandler) getSpanKeysV3(ctx context.Context, queryRangeParams *v3.QueryRangeParamsV3) (map[string]v3.AttributeKey, error) {
+	data := map[string]v3.AttributeKey{}
+	for _, query := range queryRangeParams.CompositeQuery.BuilderQueries {
+		if query.DataSource == v3.DataSourceTraces {
+			spanKeys, err := aH.reader.GetSpanAttributeKeys(ctx)
+			if err != nil {
+				return nil, err
+			}
+			return spanKeys, nil
+		}
```

Comment:
Add break statement

Reply:
Why a break statement is needed?

Reply:
Otherwise `GetSpanAttributeKeys` will run multiple times, check `getLogFieldsV3`

Reply:
There's a return statement so break is not required

Reply:
Ahh sorry